python/dgl/transform.py

```diff
@@ -673,71 +676,693 @@ def metapath_reachable_graph(g, metapath):
+                                                                                len(new_feats))
+                        new_g.edges[c_etype].data[key] = \
+                            F.cat([F.zeros(shape, F.dtype(val), F.context(val)),
+                                   new_feats], dim=0)
+        else:
+            for key, val in g.edges[c_etype].data.items():
+                new_g.edges[c_etype].data[key] = val
+
     return new_g
 
-def remove_self_loop(g):
-    """Return a new graph with all self-loop edges removed
+def remove_edges(g, eids, etype=None):
+    r"""Remove multiple edges with the specified edge type.
+    A new graph with certain edges deleted is returned.
+
+    Nodes will not be removed. After removing edges, the rest
+    edges will be re-indexed using consecutive integers from 0,
+    with their relative order preserved.
+    The features for the removed edges will be removed accordingly.
+
+    Parameters
+    ----------
+    eids : int, tensor, numpy.ndarray, list
+        IDs for the edges to remove.
+    etype : str or tuple of str, optional
+        The type of the edges to remove. Can be omitted if there is
+        only one edge type in the graph.
+
+    Return
+    ------
+    DGLHeteroGraph
+        The graph with edges deleted.
 
     Examples
-    ---------
+    --------
+    >>> import dgl
+    >>> import torch
+
+    **Homogeneous Graphs or Heterogeneous Graphs with A Single Edge Type**
+
+    >>> g = dgl.graph((torch.tensor([0, 0, 2]), torch.tensor([0, 1, 2])))
+    >>> g.edata['he'] = torch.arange(3).float().reshape(-1, 1)
+    >>> g = dgl.remove_edges(g, torch.tensor([0, 1]))
+    >>> g
+    Graph(num_nodes=3, num_edges=1,
+        ndata_schemes={}
+        edata_schemes={'he': Scheme(shape=(1,), dtype=torch.float32)})
+    >>> g.edges('all')
+    (tensor([2]), tensor([2]), tensor([0]))
+    >>> g.edata['he']
+    tensor([[2.]])
+
+    **Heterogeneous Graphs with Multiple Edge Types**
+
+    >>> g = dgl.heterograph({
+    >>>     ('user', 'plays', 'game'): (torch.tensor([0, 1, 1, 2]),
+    >>>                                 torch.tensor([0, 0, 1, 1])),
+    >>>     ('developer', 'develops', 'game'): (torch.tensor([0, 1]),
+    >>>                                         torch.tensor([0, 1]))
+    >>>     })
+    >>> g = dgl.remove_edges(g, torch.tensor([0, 1]))
+    DGLError: Edge type name must be specified
+    if there are more than one edge types.
+    >>> g = dgl.remove_edges(g, torch.tensor([0, 1]), 'plays')
+    >>> g.edges('all', etype='plays')
+    (tensor([0, 1]), tensor([0, 0]), tensor([0, 1]))
+    See Also
+    --------
+    add_nodes
+    add_edges
+    remove_nodes
+    """
+    # TODO(xiangsx): block do not support remove_edges
+    if etype is None:
+        if len(g.etypes) != 1:
+            raise DGLError('Edge type name must be specified if there are more than one ' \
+                            'edge types.')
```

Comment:
I feel this is a very common check that we should have a `utils.check_valid_etype(g, etype)` for that. Same for ntype.

Reply:
There is lots of such codes. 
I think we should handle them in a batch later.

Reply:
Sure